src/functions/operatorname.js

```diff
@@ -42,47 +38,74 @@ defineFunction({
-                    letter = letter.replace(/\u2217/, "*");
-
-                    // Use math mode for Greek letters
-                    mode = (/[\u0391-\u03D7]/.test(letter) ? "math" : "text");
-                    output.push(buildCommon.mathsym(letter, mode));
-                } else {
-                    output.push(child);
+                    let letter = child.value;
+                    if (/[\u2212\u2217]/.test(letter)) {
+                        // Per amsopn package,
+                        // change minus to hyphen and \ast to asterisk
+                        letter = letter.replace(/\u2212/, "-");
+                        letter = letter.replace(/\u2217/, "*");
+                        child.value = letter;
+                    }
                 }
             }
+            return buildCommon.makeSpan(["mop"], expression, options);
+        } else {
+            return buildCommon.makeSpan(["mop"], [], options);
         }
-        return buildCommon.makeSpan(["mop"], output, options);
     },
 
     mathmlBuilder: (group, options) => {
         // The steps taken here are similar to the html version.
-        let output = [];
-        if (group.value.value.length > 0) {
-            const temp = mml.buildExpression(
-                group.value.value, options.withFont("mathrm"));
+        let expression = mml.buildExpression(
+            group.value.value, options.withFont("mathrm"));
 
-            let word = temp.map(node => node.toText()).join("");
+        // Is expression a string or has it something like a fraction?
+        let isAllString = true;  // default
+        for (const node of expression) {
+            if (node instanceof mathMLTree.SpaceNode) {
+                // Do nothing
+            } else {
+                switch (node.type) {
+                    case "mi":
+                    case "mn":
+                    case "ms":
+                    case "mspace":
+                    case "mtext":
+                        break;  // Do nothing yet.
+                    case "mo":
+                        if (node.children.length === 1) {
+                            if (node.children[0] instanceof mathMLTree.TextNode ===
+                                false) {
+                                isAllString = false;
+                            }
+                        } else {
+                            isAllString = false;
+                        }
+                        break;
+                    default:
+                        isAllString = false;
+                }
+            }
+        }
+
+        if (isAllString) {
+            // Write a single TextNode instead of multiple nested tags.
+            let word = expression.map(node => node.toText()).join("");
+            // Per amsopn package,
+            // change minus to hyphen and \ast to asterisk
             word = word.replace(/\u2212/g, "-");
             word = word.replace(/\u2217/g, "*");
```

Comment:
What if `!isAllString`, but has these characters?

Reply:
Good point. I've made a change to pick that up.

Reply:
@ronkok If then, isn't `isAllString` unnecessary?

Reply:
`isAllString` still has a purpose. The preferred behavior is to put a string inside a single `<mo>...</mo>` tag. And that's what happens if `isAllString` remains true. When it is false, the code will write  nested tags, such as `<mo><mi>...</mi><mfrac>...</mfrac></mo>`.

Reply:
@ronkok I was thinking about these two lines.

Reply:
@ylemkimon You're right, as usual. I'll delete those two lines.